android/src/main/java/com/reactnative/googlefit/HeartrateHistory.java

```diff
@@ -157,24 +153,23 @@ protected Void doInBackground(Void... params) {
      * @param dataType DataType Fitness Data Type object
      * @param dataSourceType int Data Source Id. For example, DataSource.TYPE_RAW
      * @param value Object Values for the fitness data. They must be int or float
-     * @param startTime long Time when the fitness activity started
-     * @param endTime long Time when the fitness activity finished
+     * @param date long Time when the activity started
      * @param timeUnit TimeUnit Time unit in which period is expressed
      * @return
      */
     private DataSet createDataForRequest(DataType dataType, int dataSourceType, Double value,
-                                         long startTime, long endTime, TimeUnit timeUnit) {
+                                         long date, TimeUnit timeUnit) {
         DataSource dataSource = new DataSource.Builder()
                 .setAppPackageName(GoogleFitPackage.PACKAGE_NAME)
                 .setDataType(dataType)
                 .setType(dataSourceType)
                 .build();
 
         DataSet dataSet = DataSet.create(dataSource);
-        DataPoint dataPoint = dataSet.createDataPoint().setTimeInterval(startTime, endTime, timeUnit);
+        DataPoint dataPoint = dataSet.createDataPoint();
 
-        float f1 = Float.valueOf(value.toString());
-        dataPoint = dataPoint.setFloatValues(f1);
+        dataPoint.setTimestamp(date, timeUnit);
+        dataPoint.getValue(FIELD_BLOOD_GLUCOSE_LEVEL).setFloat(Float.valueOf(value.toString()));
```

Comment:
`Float.valueOf(value.toString())` do you mean `(float) value`

Reply:
thanks, it's much better :)